caas/kubernetes/provider/application/application.go

```diff
@@ -1362,6 +1372,15 @@ func (a *app) ApplicationPodSpec(config caas.ApplicationConfig) (*corev1.PodSpec
 		SuccessThreshold:    containerProbeSuccess,
 		FailureThreshold:    containerStartupProbeFailure,
 	}
+	// In 3.6-beta3 we moved to using startup probes with no startup delay.
+	if agentVersionNoBuild.Compare(startupProbeVersion) >= 0 {
```

Comment:
I don't quite follow why this needs a version check -- can you explain for my benefit? Why can't we just use the new values (and this will only be shipped on the 3.6 branch and later)?

Reply:
Juju controllers supports hosting other model versions.